### PR TITLE
Reorganize readme

### PR DESCRIPTION
**Pull request by erthalion**
## Description

Replace front README with more general bits and references to the
various docs section, move development hints into the how-to file. Clean
up on the way outdated Sysdig mentions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,75 +1,36 @@
...
+Welcome to the Stackrox Collector project documentation. Here you can learn
+more about idea behind the project, how to start guidelines, design overview
+and detailed references.
 
-Since the collector repo uses a submodule for the sysdig source, use
-the following command to clone.
+Collector is a component of Stackrox responsible for gathering runtime data. In
+a few words it's an agent that runs on every node under strict performance
+limitations and gather the data via kernel modules or eBPF probes (the default
+collection mode nowadays). To implement eBPF probes and collecting data the
```
Comment (ovalenti) [inline README.md:14]:
```suggestion
collection mode nowadays). To implement eBPF probes and collect data, the
```

File: README.md [hunk 2]
```diff
@@ -1,75 +1,36 @@
...
+project leverages Falco libraries via custom
```
Comment (ovalenti) [inline README.md:15]:
```suggestion
project leverages the Falco libraries via a custom
```

File: README.md [hunk 3]
```diff
@@ -1,75 +1,36 @@
...
-These instructions are for using the *JetBrains* C/C++ IDE **CLion**, but should be adaptable to any IDE that supports development over ssh/sftp.
-- If running CLion IDE for the first time:
-  - Download and install CLion (https://www.jetbrains.com/clion/download/)
-  - Create a project from the `collector/collector` directory using an existing CMakeLists file (`collector/collector/CMakeLists.txt`).
-- Create the collector builder container if not already running, or if the builder image has changed.
-  - `make start-dev`
-    - (Optional) Local builder images can used by setting the environment variable before execution using `BUILD_BUILDER_IMAGE=true make start-dev`.
-Or, builder images from a PR by with `COLLECTOR_BUILDER_TAG=<circle-build-id> make start-dev`.
+1. [How to start](docs/how-to-start.md): If you want to contribute to the
+   project, it's the best place to start. This section covers building and
```
Comment (ovalenti) [inline README.md:23]:
```suggestion
   project, this is the best place to start. This section covers building and
```

File: README.md [hunk 4]
```diff
@@ -1,75 +1,36 @@
...
+   understand how Collector works, and it's place in the grand scheme of
+   things, you may want to look here.
 
-Instructions for Linux
-- In the **File->Settings->Build, Execution, Deployment->Toolchains** window, add a new **Toolchain** entry as a **Remote Host** type.
-- Then, click in the **Credentials** section and fill out the SSH credentials used in the builder Dockerfile.
-  - Host: `localhost`, Port: `2222`, User name: `remoteuser`, Password: `c0llectah`
-- In the **File->Setting->Build, Execution, Deployment->Deployment** window click on the **Mappings** tab. Set **Deployment path** to /tmp.
-- In the **File->Settings->Build, Execution, Deployment->CMake** window add a CMake profile that uses the **Remote Host** toolchain and change **Build directory**/**Generation Path** to `cmake-build`.
+3. [Roadmap](docs/roadmap.md): Out plans and directions for future development.
```
Comment (ovalenti) [inline README.md:30]:
```suggestion
3. [Roadmap](docs/roadmap.md): Our plans and directions for future development.
```

File: README.md [hunk 5]
```diff
@@ -1,75 +1,36 @@
...
-### Building collector image(s) from the command-line
-- `make image` will create the Red Hat based collector image.
-- `make image-dev` will create a collector image based on CentOS Stream, the package manager in this image is available so additional developer tools can be installed.
-- `make image-dev-full` will extend the `image-dev` target and also compile and embed the kernel module and eBPF probes in the final image, making it easier to test changes in these components.
-
-### Setting up git hooks
-
-Git hooks are configured in the `.pre-commit-config.yaml` file using [pre-commit](https://pre-commit.com)
-In order to use them install pre-commit, and run `git config core.hooksPath ./githooks/` from the collector root directory.
+4. [References](docs/references.md): Contains comprehensive list of
```
Comment (ovalenti) [inline README.md:35]:
```suggestion
4. [References](docs/references.md): Contains a comprehensive list of
```

**Review by ovalenti — CHANGES_REQUESTED** [review]
I like the idea of moving the setup instructions out of the main README.
I added a few minor suggestions for the wording.

Only the 'Out'->'Our' spelling is blocking approval

**Review by ovalenti — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1,75 +1,36 @@
...
+a few words it is an agent that runs on every node under strict performance
+limitations and gathers the data via kernel modules or eBPF probes (the default
+collection mode nowadays). To implement eBPF probes and collect data, the
+project leverages the Falco libraries via a custom
+[fork](https://github.com/stackrox/falcosecurity-libs).
 
-```
-git clone --recurse-submodules git@github.com:stackrox/collector.git
-```
+There are opportunities to catch in many areas, so let's start.
```
Comment (JoukoVirtanen) [inline README.md:18]: What is the meaning of catch here?

File: README.md [hunk 7]
```diff
@@ -1,75 +1,36 @@
...
 
-### Development with an IDE (CLion)
+## Table of Contents
```
Comment (JoukoVirtanen) [inline README.md:20]: Put ToC at the very top.

Comment (JoukoVirtanen) [reply README.md:20]: Actually this isn't a ToC, but a list of resources.

File: README.md (hunk 6 above)
Comment (erthalion) [reply README.md]: Meaning if somebody wants to contribute, there are opportunities for that. It sounded a bit awkward, so I've changed it a bit.

**Pull request merged**